### PR TITLE
Dsl2

### PR DESCRIPTION
**Pull request by aunderwo**
More updates based on PR comments from @ggabernet . Mostly very minor
- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/bactmap/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/bactmap _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/base.config [hunk 1]
```diff
@@ -51,22 +51,22 @@ process {
...
-    memory = { check_max( 1.GB * task.attempt, 'memory' ) }
+    memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
     time   = { check_max( 1.h   * task.attempt, 'time'   ) }
   }
-  withName: BAM_SORT_SAMTOOLS:SAMTOOLS_SORT{
+  withName: SAMTOOLS_STATS {
     cpus   = { check_max( 1     * task.attempt, 'cpus'   ) }
-    memory = { check_max( 1.GB * task.attempt, 'memory' ) }
+    memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
     time   = { check_max( 1.h   * task.attempt, 'time'   ) }
```
Comment (ggabernet) [inline conf/base.config:70]: As these three resource requirements are exactly the same as the `process_low` requirements, they can be removed, and it is only necessary to specify the `label 'process_low'` inside those processes.

File: main.nf [hunk 2]
```diff
@@ -109,11 +117,10 @@ include { BWA_MEM   } from './modules/nf-core/software/bwa/mem/main'   addParams
 /* --           RUN MAIN WORKFLOW              -- */
 ////////////////////////////////////////////////////
 
-// Info required for completion email and summary
+// Info required for completion email and summary   
 def multiqc_report = []
 
 workflow {
```
Comment (ggabernet) [inline main.nf:123]: As mentioned in the previous review, and I see Harshil also commented it now on the #bactmap channel, all this part inside the `workflow {}` can be moved into a new file called `bactmap.nf` in the main pipeline directory. That leaves the `main.nf` with only the boilerplate code

**Review by ggabernet — COMMENTED** [review]
Just a couple more comments, the other changes look good now!

File: conf/base.config (hunk 1 above)
Comment (aunderwo) [reply conf/base.config]: Resolved apart from ` withName: SAMTOOLS_SORT` which needed overriding from `process_medium` for bacterial files

**Review by ggabernet — COMMENTED** [review]
Looks great to me! Do you want to add some more changes here as well? Or should I approve so you can merge these ones first?

**Comment by aunderwo** [comment]
@ggabernet good to merge now I think

File: docs/usage.md [hunk 3]
```diff
@@ -63,13 +63,13 @@ The steps in the pipeline have sensible defaults but complete control of the arg
...
 ```bash
   'iqtree' {
-      build = false
       args = '-alrt 1000 -B 1000 -m MFP -czb'
       publish_dir = 'iqtree'
   }
 ```
 
-These could be overridden by specifying a config file by adding an argument to the command line such as `-c my_args.config`.  The contents of the file could be:
+These could be overridden by specifying a config file by adding an argument to the command line such as t `-c user.config` ([user.config](../assets/user.config)).
```
Comment (ggabernet) [inline docs/usage.md:71]:
```suggestion
These could be overridden by specifying a config file by adding an argument to the command line such as `-c user.config` ([user.config](../assets/user.config)).  
```

File: main.nf [hunk 4]
```diff
@@ -44,205 +52,18 @@ Checks.aws_batch(workflow, params)
...
-    ch_software_versions = ch_software_versions.mix(CREATE_PHYLOGENY.out.raxmlng_version.ifEmpty(null))
-
-    /*
-     * MODULE: Pipeline reporting
-     */
-    GET_SOFTWARE_VERSIONS ( 
-        ch_software_versions.map { it }.collect()
-    )
-
+    include { BACTMAP } from './workflows/bactmap'
```
Comment (ggabernet) [inline main.nf:65]:
```suggestion
    include { BACTMAP } from './workflows/bactmap' addParams( summary_params: summary_params )
```

File: workflows/bactmap.nf [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    ch_software_versions = ch_software_versions.mix(CREATE_PHYLOGENY.out.raxmlng_version.ifEmpty(null))
+
+    /*
+     * MODULE: Pipeline reporting
+     */
+    GET_SOFTWARE_VERSIONS ( 
+        ch_software_versions.map { it }.collect()
+    )
+
+}
```
Comment (ggabernet) [inline workflows/bactmap.nf:185]:
```suggestion
}

////////////////////////////////////////////////////
/* --              COMPLETION EMAIL            -- */
////////////////////////////////////////////////////

workflow.onComplete {
    Completion.email(workflow, params, params.summary_params, projectDir, log, multiqc_report)
    Completion.summary(workflow, params, log)
}

workflow.onError {
    // Print unexpected parameters - easiest is to just rerun validation
    NfcoreSchema.validateParameters(params, json_schema, log)
}
```

File: main.nf [hunk 6]
```diff
@@ -44,205 +52,18 @@ Checks.aws_batch(workflow, params)
...
+    BACTMAP()
 }
 
 ////////////////////////////////////////////////////
```
Comment (ggabernet) [inline main.nf:69]: You can remove then this Completion email statment from `main.nf` as it is included inside the workflow.

File: workflows/bactmap.nf [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+                                                                                                iqtree_options: modules['iqtree'], 
+                                                                                                raxmlng_options: modules['raxmlng']
+                                                                                            )
+
+include { find_genome_size } from '../modules/local/functions.nf'
+
+////////////////////////////////////////////////////
+/* --    IMPORT NF-CORE MODULES/SUBWORKFLOWS   -- */
+////////////////////////////////////////////////////
+def fastp_options   = modules['fastp']
```
Comment (ggabernet) [inline workflows/bactmap.nf:46]:
```suggestion
def fastp_options   = modules['fastp']
def multiqc_report = []
```

**Review by ggabernet — APPROVED** [review]
Great! Just one more minor thing: generally the completion email statement is also part of the `bactmap.nf` in the dsl2 pipelines.

Then it is good to go in my opinion!

**Pull request merged**